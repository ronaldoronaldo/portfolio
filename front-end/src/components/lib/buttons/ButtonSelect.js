import React from 'react'
import { ButtonSelectStyle, SelectIcon } from './ButtonSelect.style'
import PropTypes from 'prop-types'

const ButtonSelect = ({
  size,
  children,
  text,
  color,
  onClick,
  style,
  bgColor,
  iconDirection,
  label,
  active,
  labelColor,
  ...rest
}) => {
  return (
    <ButtonSelectStyle
      style={style}
      onClick={onClick}
      {...rest}
      active={active}
      labelColor={labelColor}
    >
      <span>{label}</span>
      {active && text}
      <SelectIcon iconDirection={iconDirection} />
    </ButtonSelectStyle>
  )
}

ButtonSelect.propTypes = {
  size: PropTypes.oneOf(['xsmall', 'small', 'medium', 'large']),
  text: PropTypes.string,
  color: PropTypes.string,
  onClick: PropTypes.func.isRequired,
  bgColor: PropTypes.string,
  flex: PropTypes.number,
  border: PropTypes.string,
  borderRadius: PropTypes.number,
  marginBottom: PropTypes.bool,
  outline: PropTypes.string,
  iconDirection: PropTypes.bool.isRequired,
  label: PropTypes.string.isRequired,
  active: PropTypes.bool.isRequired
}

export default ButtonSelect
