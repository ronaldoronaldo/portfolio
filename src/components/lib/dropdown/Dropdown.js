import React, { useState } from 'react'
import {
  SelectItemStyle,
  SelectContainerStyle,
  SelectContainerHeader,
  TitleContainer,
  ActionsButtonsContainer,
  InputSearchStyle,
  ContainerIcon,
  ContainerImage,
  Image
} from './Dropdown.style'
import { IconButtonStyled } from 'styles/common/Common.styles'
import PropTypes from 'prop-types'
import FontIcon from 'components/lib/icons/FontIcon'
import { colors, fontWeights, spacing } from 'config/ui'
import Checkbox from 'components/admin/user-management-panel/checkbox'
import Button from 'components/lib/buttons/Button'

const Dropdown = ({
  onChange,
  items,
  style,
  searchInputLabel,
  mobileStyle,
  headerTitle,
  active,
  closeDropdown,
  iconNext,
  checkbox,
  actionsButtons,
  handleCancel,
  handleAdd,
  checked,
  itemsColor,
  ...rest
}) => {
  const [filteredItems, setFilteredItems] = useState(items)
  const [searchValue, setSearchValue] = useState('')
  const [itensChecked, setItensChecked] = useState([])

  const selectItems = () => {
    return filteredItems.map((item, index) => {
      if (!item) return

      if (checked && checked.indexOf(item.text || item) !== -1) {
        return
      }
      return (
        <SelectItemStyle
          key={index}
          {...rest}
          checkbox={checkbox}
          onClick={() => itemClicked(item)}
          color={itemsColor}
          iconItems={!!item.icon}
          itemDisabled={item.itemDisabled}
          image={item.image}
        >
          {checkbox && (
            <Checkbox
              checked={itensChecked.indexOf(item) !== -1}
              onChange={() => console.log('checkbox change')}
            />
          )}
          {item.icon ? (
            <ContainerIcon>
              <FontIcon
                iconName={item.icon}
                size={16}
                sizeMobile={16}
                color={colors.black5}
                style={{
                  fontWeight: fontWeights.bold,
                  marginRight: spacing.small
                }}
              />
              {item.text}
            </ContainerIcon>
          ) : item.image ? (
            <ContainerImage>
              <Image src={item.image} disabled={item.itemDisabled} />
              {item.text}
            </ContainerImage>
          ) : (
            item
          )}
          {iconNext && (
            <FontIcon iconName="caret-right" size={16} color={colors.black2} />
          )}
        </SelectItemStyle>
      )
    })
  }

  const itemClicked = item => {
    if (item.onClick) {
      item.onClick()
      return
    }
    const newArrItensChecked = itensChecked.filter(
      itemChecked => itemChecked !== item
    )
    setSearchValue('')
    onChange(item)
    !checkbox && closeDropdown()

    if (checkbox && itensChecked.indexOf(item) !== -1) {
      setItensChecked(newArrItensChecked)
    } else {
      checkbox && setItensChecked([...itensChecked, item])
    }
  }

  const searchItems = value => {
    setSearchValue(value)
    const tempItems = filteredItems.filter(item =>
      item.match(new RegExp(value, 'i'))
    )
    if (!value.length) {
      return setFilteredItems(items)
    }
    setFilteredItems(tempItems)
  }

  const closeSelectOptionsContainer = () => {
    setSearchValue('')
    closeDropdown()
  }

  return (
    <React.Fragment>
      {active && (
        <SelectContainerStyle
          mobileStyle={mobileStyle}
          style={style}
          checkbox={checkbox}
          {...rest}
        >
          <SelectContainerHeader mobileStyle={mobileStyle}>
            <TitleContainer>
              <p style={{ fontSize: 16, fontWeight: 'bold' }}>{headerTitle}</p>
              <FontIcon
                iconName={'cancel'}
                size={14}
                sizeMobile={14}
                color={colors.black3}
                onClick={closeSelectOptionsContainer}
                style={{ cursor: 'pointer' }}
              />
            </TitleContainer>
            <InputSearchStyle
              id={'dropdownSearchInput'}
              value={searchValue}
              placeholder={searchInputLabel}
              onChange={searchItems}
              onClose={closeSelectOptionsContainer}
              size={mobileStyle && checkbox && 'medium'}
              checkbox={checkbox}
            />
          </SelectContainerHeader>
          {filteredItems && selectItems()}
          {actionsButtons && (
            <ActionsButtonsContainer checkbox={checkbox}>
              <IconButtonStyled
                text="Cancelar"
                color={colors.black4}
                onClick={handleCancel}
                style={{
                  width: 95,
                  height: 40,
                  marginLeft: 8,
                  border: 'none'
                }}
                iconName=""
              />
              <Button
                text="Adicionar"
                primary
                onClick={() => handleAdd(itensChecked)}
                iconSize={16}
                style={{
                  width: 107,
                  height: 40,
                  marginLeft: 8,
                  marginRight: 16,
                  borderRadius: 10
                }}
              />
            </ActionsButtonsContainer>
          )}
        </SelectContainerStyle>
      )}
    </React.Fragment>
  )
}

Dropdown.propTypes = {
  items: PropTypes.array.isRequired,
  searchInputLabel: PropTypes.string,
  onChange: PropTypes.func,
  mobileStyle: PropTypes.bool,
  headerTitle: PropTypes.string,
  active: PropTypes.bool,
  closeDropdown: PropTypes.func,
  checkbox: PropTypes.bool,
  actionsButtons: PropTypes.bool,
  handleAdd: PropTypes.func,
  handleCancel: PropTypes.func
}

export default Dropdown
