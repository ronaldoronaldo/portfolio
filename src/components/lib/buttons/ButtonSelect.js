import React from 'react'
import { ButtonSelectStyle, SelectIcon } from './ButtonSelect.style'

const ButtonSelect = ({
  size,
  children,
  text,
  color,
  onClick,
  style,
  bgColor,
  iconDirection,
  label,
  active,
  labelColor,
  noMarginLeft,
  ...rest
}) => {
  return (
    <ButtonSelectStyle
      style={style}
      onClick={onClick}
      {...rest}
      active={active}
      labelColor={labelColor}
      noMarginLeft={noMarginLeft}
    >
      <span>{label}</span>
      {active && text}
      <SelectIcon iconDirection={iconDirection} />
    </ButtonSelectStyle>
  )
}

export default ButtonSelect
