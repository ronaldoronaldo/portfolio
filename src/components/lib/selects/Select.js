import React, { useState } from 'react'
import { ButtonSelect } from 'components/lib/buttons'
import PropTypes from 'prop-types'
import { Dropdown } from 'components/lib/dropdown'
import ClickOutside from 'utils/ClickOutside'

const Select = ({
  onChange,
  items,
  text,
  style,
  searchInputLabel,
  mobileStyle,
  headerTitle,
  label,
  labelColor,
  activeSelectButton,
  iconNext,
  marginBottom,
  dropdownStyle,
  noMarginLeft,
  ...rest
}) => {
  const [active, setActive] = useState(false)

  const onClickOutside = () => {
    if (mobileStyle) return
    setActive(false)
  }

  return (
    <React.Fragment>
      <ClickOutside onClickOutside={onClickOutside}>
        <ButtonSelect
          style={style}
          text={text}
          label={label}
          labelColor={labelColor}
          active={activeSelectButton}
          onClick={() => setActive(!active)}
          iconDirection={active}
          marginBottom={marginBottom}
          noMarginLeft={noMarginLeft}
        />
      </ClickOutside>
      {items.length > 0 && (
        <Dropdown
          items={items}
          active={active}
          onChange={onChange}
          iconNext={iconNext}
          searchInputLabel={searchInputLabel}
          mobileStyle={mobileStyle}
          headerTitle={headerTitle}
          closeDropdown={() => setActive(false)}
          style={dropdownStyle}
          {...rest}
        />
      )}
    </React.Fragment>
  )
}

Select.propTypes = {
  items: PropTypes.array.isRequired,
  text: PropTypes.string.isRequired,
  searchInputLabel: PropTypes.string,
  onChange: PropTypes.func,
  mobileStyle: PropTypes.bool,
  headerTitle: PropTypes.string,
  label: PropTypes.string,
  activeSelectButton: PropTypes.bool,
  itemsSchoolYear: PropTypes.bool
}

export default Select
