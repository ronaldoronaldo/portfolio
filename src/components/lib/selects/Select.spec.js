import React from 'react'
import { shallow, mount } from 'enzyme'
import Select from './Select'
import ButtonSelect from 'components/lib/buttons/ButtonSelect'
import Dropdown from 'components/lib/dropdown/Dropdown'

describe('<ButtonSelect />', () => {
  let props
  beforeEach(() => {
    props = {
      items: [
        'Ensino infantil',
        '1ยบ ano',
        '2ยบ ano',
        '3ยบ ano',
        '4ยบ ano',
        'item 2',
        'item 1',
        'item 2',
        'item 1',
        'item 2',
        'item 1',
        'item 2'
      ],
      text: 'Selecione o estado',
      label: 'test',
      activeSelectButton: false
    }
  })

  it('should render correctly', () => {
    const wrapper = shallow(<Select {...props} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('should show options', () => {
    const wrapper = mount(<Select {...props} active={true} />)
    wrapper.find(ButtonSelect).simulate('click')
    expect(wrapper.contains({ ...props })).toBe(true)
  })

  it('should add and remove options from screen', () => {
    const wrapper = mount(<Select {...props} active={false} />)
    const button = wrapper.find(ButtonSelect)
    button.simulate('click')
    const dropdown = wrapper.find(Dropdown)
    setTimeout(() => expect(dropdown.prop('active')).toBe(true), 300)
    button.simulate('click')
    setTimeout(() => expect(dropdown.prop('active')).toBe(false), 300)
  })
})
