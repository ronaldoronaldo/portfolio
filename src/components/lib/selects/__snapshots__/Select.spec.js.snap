// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ButtonSelect /> should render correctly 1`] = `ShallowWrapper {}`;
